workspacemanager/WorkspaceManagerPresenterTest.py

```diff
@@ -0,0 +1,173 @@
+from mock.mock import self
+
+from WorkspaceManagerPresenter import WorkspaceManagerPresenter
+from WorkspaceProvider import WorkspaceProvider
+from WorkspaceView import WorkspaceView
+from command import Command
+import unittest
+import mock
+from mock import call
+
+
+#ToDo askOwen, view related tests, i.e view returns 'selected_workspace' that does not exist
+#TODO handle mantid load fail inquiry
+#TODO Test constructor and make this test specific
+
+class WorkspaceManagerPresenterTest(unittest.TestCase):
+    def setUp(self):
+        self.workspace_provider = mock.create_autospec(spec=WorkspaceProvider)
+        self.view = mock.create_autospec(spec=WorkspaceView)
+
+
+
+    def test_load_one_workspace(self):
+        self.presenter = WorkspaceManagerPresenter(self.view, self.workspace_provider)
+        self.view.get_workspace_to_load_path = mock.Mock(return_value=r'C:\a\b\cde.nxs')
+        self.workspace_provider.getWorkspaceNames = mock.Mock(return_value=['cde'])
+        self.presenter.notify(Command.LoadWorkspace)
+        self.view.get_workspace_to_load_path.assert_called_once()
+        self.workspace_provider.Load.assert_called_with(Filename=r'C:\a\b\cde.nxs', OutputWorkspace='cde')
+        self.view.display_loaded_workspaces.assert_called_with(['cde']) #TODO could this be has is_called?
+
+    def test_load_multiple_workspaces(self):
+        self.presenter = WorkspaceManagerPresenter(self.view, self.workspace_provider)
+        self.view.get_workspace_to_load_path = mock.Mock(
+            side_effect=[r'C:\path\to\file1.nxs', r'C:\path\to\any\file2.nxs', r'C:\path\to\file3.nxs'])
+        self.workspace_provider.getWorkspaceNames = mock.Mock(side_effect=[['file1'], ['file1', 'file2'],
+                                                                           ['file1', 'file2', 'file3']])
+        for i in range(3):
+            self.presenter.notify(Command.LoadWorkspace)
+        load_calls = [call(Filename='C:\\path\\to\\file1.nxs', OutputWorkspace='file1'),
+                      call(Filename='C:\\path\\to\\any\\file2.nxs', OutputWorkspace='file2'),
+                      call(Filename='C:\\path\\to\\file3.nxs', OutputWorkspace='file3')]
+        self.workspace_provider.Load.assert_has_calls(load_calls)
+        #The test should just finish here
+        # <Not Maybe not useful> TODO askOwen case1
+        display_workspace_calls = [call(['file1']), call(['file1', 'file2']), call(['file1', 'file2', 'file3'])]
+        self.view.display_loaded_workspaces.assert_has_calls(display_workspace_calls)
+        # </Maybe not useful>
+
+    # TODO make sure that workspace provider is not called on failure cases
+    def test_rename_workspace(self):
```

Comment:
You have a lot of "Happy Path" test cases, but very few "Destructive Test Cases". Destructive test cases are more useful. Rename your existing test cases to things like `test_rename_workspace_succeeds` and then add tests for cases in which the operation cannot succeed for various reasons and add test cases for those `test_rename_workspace_throws_when_X` where X is some failure scenario. I estimate that you should have about a 2:1 ratio (or greater) between destructive and happy-path test cases.